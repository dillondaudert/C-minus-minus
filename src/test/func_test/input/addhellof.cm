int i, j, k, l;
float x1, x2;

int main () {
	
	write('Hello, world');
	write('Hello, again :)');
	i=1;  k=3; l=4;
	j = i + k + l;
	write(j);
	exit; 

}
