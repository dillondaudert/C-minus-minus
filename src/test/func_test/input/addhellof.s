	.section	.rodata
.int_wformat: .string "%d\n"
.str_wformat: .string "%s\n"
.int_rformat: .string "%d"
	.comm _gp, 32, 8

.string_const0: .string "Hello, world"
.string_const1: .string "Hello, again :)"

	.globl main
	.type main @function

main:	nop
	pushq	%rbp
	movq	%rsp, %rbp
	movl	$.string_const0, %ebx
	movl	%ebx, %esi
	movl	$0, %eax
	movl	$.str_wformat, %edi
	call	printf
	movl	$.string_const1, %ebx
	movl	%ebx, %esi
	movl	$0, %eax
	movl	$.str_wformat, %edi
	call	printf
	leave
	ret
	.size	main, .-main

