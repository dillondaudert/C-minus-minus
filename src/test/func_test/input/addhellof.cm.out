st_init created the symbol table hash map
Added i to $$names, count=1
Adding new symbol: i, _gp, offset: 0, type: 0, size: 0, arr: 0
st_add added i at position 72
Added j to $$names, count=2
Adding new symbol: j, _gp, offset: 1, type: 0, size: 0, arr: 0
st_add added j at position 75
Added k to $$names, count=3
Adding new symbol: k, _gp, offset: 2, type: 0, size: 0, arr: 0
st_add added k at position 74
Added l to $$names, count=4
Adding new symbol: l, _gp, offset: 3, type: 0, size: 0, arr: 0
st_add added l at position 77
updating symbol i
Symbol 72:i found
Symbol i updated with size 4, offset 0
updating symbol j
Symbol 75:j found
Symbol j updated with size 4, offset 4
updating symbol k
Symbol 74:k found
Symbol k updated with size 4, offset 8
updating symbol l
Symbol 77:l found
Symbol l updated with size 4, offset 12

Added x1 to $$names, count=1
Adding new symbol: x1, _gp, offset: 0, type: 0, size: 0, arr: 0
st_add added x1 at position 400
Added x2 to $$names, count=2
Adding new symbol: x2, _gp, offset: 1, type: 0, size: 0, arr: 0
st_add added x2 at position 155
updating symbol x1
Symbol 400:x1 found
Symbol x1 updated with size 8,offset 16
updating symbol x2
Symbol 155:x2 found
Symbol x2 updated with size 8,offset 24


Adding new symbol: main, main, offset: 0, type: 2, size: 4, arr: 0
st_add added main at position 0
Added process main to stable


Adding new symbol: .string_const0, .string_const0, offset: 0, type: 0, size: 13, arr: 0
st_add, added .string_const0 to collision LL at 0

Adding new symbol: .string_const1, .string_const1, offset: 0, type: 0, size: 16, arr: 0
st_add, added .string_const1 to collision LL at 0







Static: 	.comm _gp, 32, 8

Freed symbol at position 0
Freed LL symbol at position 0
Freed link in collision LL at 0
Freed LL symbol at position 0
Freed link in collision LL at 0
Freed symbol at position 72
Freed symbol at position 74
Freed symbol at position 75
Freed symbol at position 77
Freed symbol at position 155
Freed symbol at position 400
Freed symbol table
